### ISSUE TITLE
Feature Request - "Don't download unless a release matching preferred tags is found"

### ISSUE THREAD

**Issue by kniffs** [Radarr/Radarr#562]
I thought about new releases and came up with this.

Generally because most Blurays are released around the same time, Radarr will grab the first one to match the quality (Bluray - 1080p) and abides by size thresholds you have set.

As an example, when it is then later released with a DTS soundtrack, or by another group that you have set in "preferred tags" - it will not be upgraded because the quality already matches the release you already have and radarr will discard it as 'better'.

Thoughts?

An alternative method could be "let preferred tags upgrade movies of same quality"

**Comment by Crowley007** [comment]
Currently you can do this by setting "must contain" words as a tag in the indexers page. You have to then set the tag for the movies you want. 

I use this to get the remux releases. Of course, as you said this means you cant get a lower quality release and upgrade it.

**Comment by kniffs** [comment]
That is true.... But it would be nice to always have the best quality available at every step until all of the requirements are met.

By setting a "must contain" word like DTS it might take a week to have Arrival show up on plex at all for instance.

**Comment by Crowley007** [comment]
I know, the "1080p Blu-ray" contains way too much variance in quality to be considered just one quality.  

I have requested that Remux be added as a quality, but there might be other ways solve this. Custom qualities with their own required words and size limits would probably cover most usage cases.

**Comment by kniffs** [comment]
Yeah that would be awesome. Maybe it would be too easy to mess up for most users though, if you can add any word as a requirement.

**Comment by Crowley007** [comment]
That's what "Advanced Settings" are for.  ;)

**Comment by scuppasteve** [comment]
I came here exactly for this same request. I think the "must contain" is the wrong solution for most of this. They could just have a final cutoff that includes "+ Preferred tag"

**Comment by bazbjzy** [comment]
I agree.. I generally want first HD avail, then 1080p encode, then Bluray Remux. I have tags setup for "must contain" but I have to wait to add them to the movie until it has picked up one of the lesser HD versions or it'll sit around waiting for a remux (which sometimes takes a few more days to come out)

I like the suggestion of "let preferred tags upgrade movies of same quality". This would allow me to set all my new movies to a profile where it will download any HD and have a Bluray 1080p cutoff. Then I'd have "Remux" in the preferred tags and it should grab a Remux even though I already have a 1080p version.

**Comment by geogolem** [comment]
well.. maybe must contain is ok.. as it is.. but it is solving a different problem..

the current must contain refers to whether a release is downloaded or not..

the must contain you guys are referring to, is whether a cut off is met or not... they are distinct problems .. and i think both are useful.

**Comment by bobbintb** [comment]
There is a lot of variance on issues like this, depending on user preference. I've found this with Sonarr. For some, this can be accomplished with a variety of settings. I know it might not be an option for some, due to technical skill, but custom scripts are an option for many of these issues. As far as Sonar went, the developers are understandably hesitant to add things like this as it makes the system more complicated and difficult to support for little return to most users.

To be honest, a lot of these issues can be solved with customs scripts but not everyone has the skill. I wish there was more community involvement as far as that goes. I agree, the quality specifications are lacking and I have discussed these sorts of issues many times with the Sonarr devs about it but to be honest, I don't know if there can be a better that doesn't overcomplicate things for others.

I would suggest looking into the custom scripts. They can be quite powerful. For example, I was looking for a way for more granularity in the 1080phdtv category in Sonarr. DIMENSION releases are great and usually better than bluray. I only wanted 1080phdtv rips if they were DIMENSION and there wasn't a way to accomplish this as is but it is doable with a custom on-grab script. Custom scripts are a great way to personalize and extend flexibility without succumbing to feature creep.

**Comment by Crowley007** [comment]
Adding the Remux quality has solved my problems. Also setting the preferred words to DTS-HD MA etc usually results in best available 1080p Blu-ray rip being snatched when no Remux is available.

**Comment by rustymyers** [comment]
I ran into trouble working around this today. I have the 'must include' indexers set to require "hevc", but the profile set on movies does not enforce this. The profile I have created uses this tag, but it's only option is to set it as a 'Preferred tag', which means it's not enforced. After some troubleshooting I found that I can force movies to require the "hevc" enforcement by setting the 'Tags' field. Unfortunately, there isn't a bulk-edit for tags. I created this script to update all my movies with the "hevc" tag, and now it seems to behaving as I'd expect. 

My suggestion would be to include a 'must include' tags along side the 'preferred tag' in the Profiles. This way I could enforce a restriction without having to tag every movie. An alternate would be bulk editing of tags, which I should request too.
